api/server/router/swarm/cluster_routes.go

```diff
@@ -232,6 +232,13 @@ func (sr *swarmRouter) createService(ctx context.Context, w http.ResponseWriter,
 		}
 	}
 
+	if versions.LessThan(version, "1.46") {
+		if service.TaskTemplate.ContainerSpec != nil && service.TaskTemplate.ContainerSpec.OomScoreAdj != 0 {
+			// OomScoreAdj was added in API 1.46
+			service.TaskTemplate.ContainerSpec.OomScoreAdj = 0
+		}
+	}
```

Comment:
Is this function the right place for this code? In https://github.com/moby/moby/pull/46386 the equivalent change was made in `func adjustForAPIVersion`, which is called by both `func createService` and `func updateService`. (I think above code block for `Healthcheck.StartInterval` is also misplaced.)

Reply:
I'm not certain, I'm following the pattern that [already exists](https://github.com/moby/moby/blob/a0d343a4b517e141c8065c09d287e56f761ef4e5/api/server/router/swarm/cluster_routes.go#L228) in that function. I haven't read the entire codebase yet.

Reply:
I asked @dperny: unfortunately, the pattern you copied was buggy code. I have opened a PR to address that deficiency. https://github.com/moby/moby/pull/47991

You will need to move the gating of `OomScoreAdj` by API version into `func adjustForAPIVersion` so that it applies to both create and update operations. And you will have to update version-history.md to mention that support was added to both the create and update API endpoints.

Reply:
I read through that code and debugged it manually. It appears as if it should return an error instead.  What should the intended behavior be?